Unit Tests/Tests/Buffer/ConcreteBufferTests.swift

```diff
@@ -0,0 +1,503 @@
+        let dependencies = DependenciesDoubles(store: stubStore)
+        
+        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
+        sut.delegate = stubDelegate
+        
+        XCTAssertEqual(sut.currentState, initialState)
+        XCTAssertEqual(stubDelegate.didUpdateState_actualCallCount, 0)
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        stubStore.report(subsequentState)
+        
+        /******************/
+        /*----- THEN -----*/
+        /******************/
+        
+        let expectedState = initialState
+        
+        XCTAssertEqual(sut.currentState, expectedState)
+        XCTAssertEqual(stubDelegate.didUpdateState_actualCallCount, 0)
+        XCTAssertNil(stubDelegate.didUpdateState_stateLastReceived)
+    }
+    
+    func test_didUpdateState_shouldFilterOutUnsupportedSignatures() {
+        /******************/
+        /*---- GIVEN -----*/
+        /******************/
+        
+        let initialState = VersionedState(
+            chatState: .empty,
+            auxiliaryState: .empty,
+            version: 1,
+            signature: .initialState
+        )
+        
+        let subsequentState = VersionedState(
+            chatState: .empty,
+            auxiliaryState: .empty,
+            version: 2,
+            signature: .subscriptionStateUpdated
+        )
+        
+        let stubStore = StubStore(state_toReturn: initialState, register_expectedCallCount: 1, unregister_expectedCallCount: 1)
+        let stubFilter = StubStateFilter(hasModifiedSubstate_defaultValueToReturn: true,
+                                         hasCompleteSubstate_defaultValueToReturn: true,
+                                         hasSupportedSignature_valuesToReturn: [.initialState : true,
+                                                                                .subscriptionStateUpdated : false])
+        let stubDelegate = StubBufferDelegate(didUpdateState_expectedCallCount: 0)
+        
+        let dependencies = DependenciesDoubles(store: stubStore)
+        
+        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
+        sut.delegate = stubDelegate
+        
+        XCTAssertEqual(sut.currentState, initialState)
+        XCTAssertEqual(stubDelegate.didUpdateState_actualCallCount, 0)
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        stubStore.report(subsequentState)
+        
+        /******************/
+        /*----- THEN -----*/
+        /******************/
+        
+        let expectedState = initialState
+        
+        XCTAssertEqual(sut.currentState, expectedState)
+        XCTAssertEqual(stubDelegate.didUpdateState_actualCallCount, 0)
+        XCTAssertNil(stubDelegate.didUpdateState_stateLastReceived)
+    }
+    
+    func test_didUpdateState_shouldUseUnsupportedSignaturesAsInitialState() {
```

Comment:
I have added all 4 tests mentioned above.

Reply:
Ace, thank you.  With the addition of these I'm not sure we need the following tests any longer?
- `test_init_withIncompleteState_hasNilCurrentState`
- `test_init_withCompleteState_hasCurrentState`
- `test_init_registersAsStoreListener`

Reply:
I've deleted the following tests:

- `test_init_withIncompleteState_hasNilCurrentState`
- `test_init_withCompleteState_hasCurrentState`

The implementation of `test_init_registersAsStoreListener` looks similar to other tests, but because it's main purpose is to assert the registration process I think it would be beneficial to keep it in the codebase.

Reply:
Sounds good, thanks 👍